packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/MMDeepLinkConnectUI.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { openWindow } from "../../../../../utils/web/openWindow.js";
+import type { InjectedSupportedWalletIds } from "../../../../../wallets/__generated__/wallet-ids.js";
+import type { Wallet } from "../../../../../wallets/interfaces/wallet.js";
+import type { WalletInfo } from "../../../../../wallets/wallet-info.js";
+import { wait } from "../../../../core/utils/wait.js";
+import type { InjectedWalletLocale } from "../../../wallets/injected/locale/types.js";
+import { ConnectingScreen } from "../../../wallets/shared/ConnectingScreen.js";
+
+/**
+ * @internal
+ */
+export const MMDeepLinkConnectUI = (props: {
+  onGetStarted: () => void;
+  locale: InjectedWalletLocale;
+  wallet: Wallet<InjectedSupportedWalletIds>;
+  walletInfo: WalletInfo;
+  onBack?: () => void;
+}) => {
+  const [errorConnecting, setErrorConnecting] = useState(false);
+  const locale = props.locale;
+
+  const handleConnect = useCallback(async () => {
+    try {
+      connectPrompted.current = true;
+      setErrorConnecting(false);
+      await wait(1000);
```

Comment:
Just curious, why the pause here?

Reply:
To prevent immediately redirecting the user to wallet app which could be jarring. Showing the spinner for at least 1 second makes it a bit nicer

Reply:
Makes sense

Reply:
( I've since removed this and changed it to open the link on click )